timescaledb/how-to-guides/schema-management/json.md

```diff
@@ -13,51 +12,49 @@ CREATE TABLE metrics (
-JSONB structure and stored as columns. This is because field accesses are more
-efficient on table columns than inside of JSONB structures. Storage is also more
-efficient.
-
-2. We use the JSONB data type (that is, JSON stored in a binary format) and not the JSON data type. JSONB data types are
-more efficient in both storage overhead and lookup performance.
 
-<highlight type="tip">
-Often, people use JSON for sparse data as opposed
-to user-defined data. We do not recommend this usage inside TimescaleDB for most
-datasets (unless the data is extremely sparse, e.g., more than 95% of fields for
-a row are empty). Instead, we suggest using NULLable fields and, if possible,
-running on top of a compressed file system like ZFS.
+When you are defining a schema using JSON, ensure that common fields, such as
+`time`, `user_id`, and `device_id`, are pulled outside of the JSONB structure
+and stored as columns. This is because field accesses are more efficient on
+table columns than inside of JSONB structures. Storage is also more efficient.
+
+You should also use the JSONB data type, that is, JSON stored in a binary
+format, rather than JSON data type. JSONB data types are more efficient in both
+storage overhead and lookup performance.
+
+<highlight type="note">
+Use JSONB for user-defined data rather than sparse data. This works best for most
+data sets. For sparse data, use NULLable fields and, if possible, run on top of
+a compressed file system like ZFS. This will work better than a JSONB data type,
+unless the data is extremely sparse, for example, more than 95% of fields for a
+row are empty.
 </highlight>
 
-
-### Indexing the entire JSONB structure [](indexing-all-json)
-
-When indexing JSONB data across all fields that may be contained inside, it is
-often best to use a GIN index. PostgreSQL documentation has
-a [nice description][json-indexing] of the different types of GIN indexes
-available on JSON data. If in doubt, it is best to use the default GIN operator
-since it allows for more powerful queries:
+## Index the JSONB structure
+When you index JSONB data across all fields, it is usually best to use a GIN
+(generalized inverted) index. In most cases, you can use the default GIN
+operator, like this:
 
 ```sql
 CREATE INDEX idxgin ON metrics USING GIN (data);
 ```
 
-Please note that this index only optimizes queries for which the WHERE clause
-uses the `?`, `?&`, `?|`, or `@>` operator (for a description of these operators
-see [this table][json-operators] in the PostgreSQL docs). So you should make
-sure to structure your queries appropriately.
+For more information about GIN indexes, see the
+[PostgreSQL documentation][json-indexing].
 
-### Indexing individual fields within a JSONB [](indexing-json-fields)
+This index only optimizes queries where the `WHERE` clause uses the `?`, `?&`,
+`?|`, or `@>` operator. For more information about these operators, see the
```

Comment:
```suggestion
`?|`, or `@>` operators. For more information about these operators, see the
```
I think? Does an 'or' list take the plural?

Reply:
Verb needs to agree with the part of the subject that is near to it. By my calculation, then, we need to make "uses" and "operator" agree. So I think it's singular, but I'm happy to be proven wrong here 😉

Reply:
I agree! I don't think that's the rule I was confused about, since 'uses' needs to agree with 'clause' and not 'operator'. But reading it again I think it should be 'operator' in the singular. I was confused because this is really a shorthand for "the `?` operator, `?&` operator,
`?|` operator, or `@>` operator". But you are right! :)